ci/do_ci.sh

```diff
@@ -361,32 +356,28 @@ elif [[ "$CI_TARGET" == "check_format" ]]; then
   ./tools/code_format/check_format.py check
   ./tools/code_format/format_python_tools.sh check
   ./tools/proto_format/proto_format.sh check --test
-  exit 0
 elif [[ "$CI_TARGET" == "check_repositories" ]]; then
   echo "check_repositories..."
   ./tools/check_repositories.sh
-  exit 0
 elif [[ "$CI_TARGET" == "check_spelling" ]]; then
   echo "check_spelling..."
   ./tools/spelling/check_spelling.sh check
-  exit 0
 elif [[ "$CI_TARGET" == "fix_spelling" ]];then
   echo "fix_spell..."
   ./tools/spelling/check_spelling.sh fix
-  exit 0
 elif [[ "$CI_TARGET" == "check_spelling_pedantic" ]]; then
   echo "check_spelling_pedantic..."
   ./tools/spelling/check_spelling_pedantic.py --mark check
-  exit 0
 elif [[ "$CI_TARGET" == "fix_spelling_pedantic" ]]; then
   echo "fix_spelling_pedantic..."
   ./tools/spelling/check_spelling_pedantic.py fix
-  exit 0
 elif [[ "$CI_TARGET" == "docs" ]]; then
   echo "generating docs..."
   docs/build.sh
-  exit 0
 else
   echo "Invalid do_ci.sh target, see ci/README.md for valid targets."
   exit 1
 fi
+
+run_process_test_result
```

Comment:
We will loose some flaky result if this is done at the very end of the script. Can you move this to part of `bazel_with_collection` function?

Reply:
are you referring to situations where you have failed tests? I thought the intention was to only report flaky tests when no failed tests exist

Reply:
No, in compile time options and asan we run same test with different options, then the latter run overwrite the previous results so you lose the flaky status.

Reply:
ohh. I see. Will do!

Reply:
@lizan This has a level of complication that in asan/tsan build where they pushd into the filter_example_directory. Obviously in that directory, the processing script is not going to be available. So what I have tried is to do is:
```
call script && echo "cannot find the script"
```
and explicitly call the script when asan/tsan popd back to the envoy directory.